### PR TITLE
Spells/DK: Scourge Strike shadow dmg even if main dmg absorbed

### PR DESCRIPTION
**Pull request by ghost**
by @P-Kito; retrieved from #15027.

Scourge Strike deals shadow damage even if the physical damage was absorbed,
using SpellHitFn() and HandleBeforeHit to achieve this effect.

### CODE REVIEW

**Comment by Demonid** [comment]
Paladin Seals should do damage on absorb too, maybe this can help fix that other issue.

**Comment by robinsch** [comment]
Wrong procFlag / procEx in DB.

**Comment by P-Kito** [comment]
@robinsch  Are you sure? This is some other kind of absorb problem. Or are you referrend to the Paladin Seals?

**Comment by tkrokli** [comment]
@robinsch : have you got an example of how you think it should be instead?
(I don't know enough to see where a procFlag/procEx is used in this PR.)

**Comment by P-Kito** [comment]
This PR is actually a bit wrong. 

I found out that GetHitDamage() will give us the damage it will do without reductions. And therefore too much shadow damage.
So there must be another solution to stop the absorb.

**Comment by tkrokli** [comment]
Scourge Strike (general spell) http://wotlk.openwow.com/spell=70890 :
```
Scourge Strike
100 yd range
Instant
An unholy strike that deals 70% of weapon damage as Physical damage
plus an additional amount.
In addition, for each of your diseases on your target,
you deal an additional 12% of the Physical damage done as Shadow damage.
```
Scourge Strike rank 1: http://wotlk.openwow.com/spell=55090
```
Scourge StrikeScourge Strike			Rank 1
5 yd range
Instant
An unholy strike that deals 70% of weapon damage as Physical damage
plus 340.
In addition, for each of your diseases on your target,
you deal an additional 12% of the Physical damage done as Shadow damage.
```
Scourge Strike rank 2 : http://wotlk.openwow.com/spell=55265
Scourge Strike rank 3 : http://wotlk.openwow.com/spell=55270
Scourge Strike rank 4 : http://wotlk.openwow.com/spell=55271

**Comment by tkrokli** [comment]
@P-Kito : I think we need another multiplier to HandleBeforeHit() or HandleAfterHit(), but how to use it?
```diff
@@ -1441,10 +1441,14 @@ class spell_dk_scourge_strike : public SpellScriptLoader
             spell_dk_scourge_strike_SpellScript()
             {
                 multiplier = 1.0f;
+                damageDone = 0;
+                shadowDamage = 0.12f;
             }
 
         private:
             float multiplier;
+            int32 damageDone;
+            float shadowDamage;
 
             bool Validate(SpellInfo const* /*spellInfo*/) override
             {

**Comment by P-Kito** [comment]
nope. I remade it. What we need is a function GetCleanDamage() and from then it's easy.

I will upload soon.

**Comment by LuqJensen** [comment]
debugging the damageDone var indicates that this PR is faulty.
Combat log/recount reports physical damage hit as 331, but damageDone is 414. (dummies have armor)
In turn shadow damage of SS with 2 active diseases is 99 instead of 79.44.
TL;DR its wrong because damage reduction of target is not taken into consideration due to the execution time of BeforeHit hook.

**Comment by P-Kito** [comment]
As I said... I reworked it and implemented GetCleanDamage().
I cant upload it right now.

Please close this PR, as it's WRONG.
